apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -556,13 +538,28 @@ pub(crate) async fn process_response<T: HttpBody>(
                 Span::current().record(OTEL_STATUS_CODE, OTEL_STATUS_CODE_ERROR);
                 raw.map_error(
                     result,
-                    connector,
+                    &connector,
                     context,
                     debug_context,
                     supergraph_request,
                 )
             }
         }
+    };
+
+    if let MappedResponse::Error { ref error, .. } = mapped_response {
+        emit_error_event(
+            &error.code(),
+            &error.message,
+            Some(Path::from_response_key(&error.response_key)),
+        );
+    }
+
+    connector::request_service::Response {
+        context: context.clone(),
```

Comment:
Can `Response` receive a reference of `context`? not really sure about the implications of this clone, but as far as I remember it is a map and Response doesnt need ownership of it

Reply:
let's sequence this work in the next phase